bgpd/bgp_fsm.c

```diff
@@ -110,9 +110,9 @@ static int bgp_peer_reg_with_nht(struct peer *peer)
 	    && !CHECK_FLAG(peer->bgp->flags, BGP_FLAG_DISABLE_NH_CONNECTED_CHK))
 		connected = 1;
 
-	return bgp_find_or_add_nexthop(
-		peer->bgp, peer->bgp, family2afi(peer->su.sa.sa_family),
-		NULL, peer, connected);
+	return bgp_find_or_add_nexthop(peer->bgp, peer->bgp,
+				       family2afi(peer->su.sa.sa_family),
+				       SAFI_UNSPEC, NULL, peer, connected);
```

Comment:
wouldn't this just be SAFI_UNICAST?  This is peer registration with nht.  Clearly it has to be SAFI_UNICAST.

Reply:
I think this is NHT resolution of the peer directly and explicitly over IPv4/v6 reachability (non-labeled). Probably that's why SAFI_UNSPEC is being used. But I maybe wrong.

Reply:
The SAFI in the function is used to determine what table the associated route is a member of. In this case no table is indicated as this is a peer NH lookup and no path is provided and in fact the SAFI is not used in the function at all. The value is a placeholder in this call. I'm happy to change it to SAFI_UNICAST if that seems more correct. I set it to unspec to indicate it was an unused value.

Reply:
A path is *provided* by the peer data structure itself(hence the lookup to the correct afi from the peer structure itself).  If we are passing in safi now then we should pass in the correct value.

Reply:
Ack - done